homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -44,65 +63,77 @@ class SurePetcareBinarySensor(BinarySensorDevice):
     """A binary sensor implementation for Sure Petcare Entities."""
 
     def __init__(
-        self, _id: int, name: str, spc, device_class: str, sure_type: SureThingID
+        self: BinarySensorDevice,
+        _id: int,
+        spc: SurePetcareAPI,
+        device_class: str,
+        sure_type: SureProductID,
     ):
         """Initialize a Sure Petcare binary sensor."""
         self._id = _id
-        self._name = name
-        self._spc = spc
-        self._device_class = device_class
         self._sure_type = sure_type
-        self._state = {}
+        self._device_class = device_class
+
+        self._spc: SurePetcareAPI = spc
+        self._spc_data: Dict[str, Any] = self._spc.states[self._sure_type].get(self._id)
+        self._state: Dict[str, Any] = {}
+
+        _LOGGER.debug("%s -> self._spc_data: %s", self._id, self._spc_data)
```

Comment:
We don't want side effects in init method. Please remove or move the logging.

Reply:
Thats just for debugging as its WIP, will remove this (and probably other logging currently in the code, also the commented code) before removing the "WIP" :)